plugins/spi/SPIOutputTest.cpp

```diff
@@ -738,3 +739,233 @@ void SPIOutputTest::testCombinedAPA102Control() {
+                                0xFF, 0, 0, 0,  // Pixel 12
+                                0xFF, 0, 0, 0,  // Pixel 13
+                                0xFF, 0, 0, 0,  // Pixel 14
+                                0xFF, 0, 0, 0,  // Pixel 15
+                                0xFF, 0, 0, 0,  // Pixel 16
+                                0};
+  OLA_ASSERT_DATA_EQUALS(EXPECTED8, arraysize(EXPECTED8), data, length);
+  OLA_ASSERT_EQ(5u, backend.Writes(0));
+
+  // test9
+  // create new output with pixel_count=17 and check data length
+  // setup pixel_count to 17
+  options.pixel_count = 17;
+  // setup SPIOutput
+  SPIOutput output3(m_uid, &backend, options);
+  // set personality
+  output3.SetPersonality(this_test_personality);
+  // generate dmx data
+  buffer.SetFromString(
+        std::string("255,0,0,0, 255,0,0,0, 255,0,0,0, 255,0,0,0, 255,0,0,0,") +
+                    "255,0,0,0, 255,0,0,0, 255,0,0,0, 255,0,0,0, 255,0,0,0," +
+                    "255,0,0,0, 255,0,0,0, 255,0,0,0, 255,0,0,0, 255,0,0,0," +
+                    "255,0,0,0, 255,0,0,0");
+  output3.WriteDMX(buffer);
+  data = backend.GetData(0, &length);
+  const uint8_t EXPECTED9[] = { 0, 0, 0, 0,
+                                0xFF, 0, 0, 0,  // Pixel 1
+                                0xFF, 0, 0, 0,  // Pixel 2
+                                0xFF, 0, 0, 0,  // Pixel 3
+                                0xFF, 0, 0, 0,  // Pixel 4
+                                0xFF, 0, 0, 0,  // Pixel 5
+                                0xFF, 0, 0, 0,  // Pixel 6
+                                0xFF, 0, 0, 0,  // Pixel 7
+                                0xFF, 0, 0, 0,  // Pixel 8
+                                0xFF, 0, 0, 0,  // Pixel 9
+                                0xFF, 0, 0, 0,  // Pixel 10
+                                0xFF, 0, 0, 0,  // Pixel 11
+                                0xFF, 0, 0, 0,  // Pixel 12
+                                0xFF, 0, 0, 0,  // Pixel 13
+                                0xFF, 0, 0, 0,  // Pixel 14
+                                0xFF, 0, 0, 0,  // Pixel 15
+                                0xFF, 0, 0, 0,  // Pixel 16
+                                0xFF, 0, 0, 0,  // Pixel 17
+                                0, 0};  // now we have two latch bytes...
+  OLA_ASSERT_DATA_EQUALS(EXPECTED9, arraysize(EXPECTED9), data, length);
+  OLA_ASSERT_EQ(6u, backend.Writes(0));
+
+  // test 10
+  // test different pixel brightness levels
+  // generate dmx data
+  buffer.SetFromString(
+        std::string("  0,0,0,0,   9,0,0,0,  16,0,0,0,  42,0,0,0,  84,0,0,0,") +
+                    "127,0,0,0, 172,0,0,0, 206,0,0,0, 247,0,0,0, 254,0,0,0," +
+                    "255,0,0,0, 255,0,0,0, 255,0,0,0, 255,0,0,0, 255,0,0,0," +
+                    "255,0,0,0, 255,0,0,0");
+  output3.WriteDMX(buffer);
+  data = backend.GetData(0, &length);
+  const uint8_t EXPECTED10[] = { 0, 0, 0, 0,
+                                0xE0 && 0, 0, 0, 0,  // Pixel 1
```

Comment:
This test can't be working with the selected personality, or there's a bigger issue.

Reply:
Add that issue is you've not added it via the CPPUNIT_TEST macro at the top of the file, so it's not actually being run, hence the tests are still passing.

Reply:
Also && here.